File: _tools/mlrs/example/LoginRq.go
package example

import (
	"log"

	"mlrs/b3"
	"mlrs/b3/config"
	"mlrs/b3/core"
	"mlrs/b3/factory"
)

func init() {
	factory.RegisterMap().Register("LoginRq", &LoginRq{})
}

// LoginRq LoginRq
type LoginRq struct {
	core.Action
}

func (n *LoginRq) Initialize(params *config.BTNodeCfg) {
	n.BaseNode.Initialize(params)
}

func (n *LoginRq) OnOpen(tick *core.Tick) {
	log.Println(n.GetName())
	tick.Blackboard.SetMem("login", true)
}

func (n *LoginRq) OnTick(tick *core.Tick) b3.Status {
	return b3.SUCCESS
}

File: _tools/mlrs/lua/node/LoginRq.lua
function OnOpen(node, tick)
    log.debug("LoginRq OnOpen , name = " .. tostring(node:GetName()) ..", title = ".. tostring(node:GetTitle()))
    tick.Blackboard:SetMem("login", true)
end

function OnTick(node, tick)
    log.debug("LoginRq OnTick , name = " .. tostring(node:GetName()) ..", title = ".. tostring(node:GetTitle()))
    return "success"
end
